TestAssignmentDoItSoftware/TestAssignmentDoItSoftware/AuthScreen/AuthScreenViewController.m

```diff
@@ -0,0 +1,339 @@
+    tfPassword.layer.borderColor = grayColor.CGColor;
+
+    btnAuth.layer.cornerRadius = 2.0;
+    btnAuth.layer.borderWidth = 1.0;
+    
+    [[NSNotificationCenter defaultCenter] removeObserver:self name:@"registrationfailed" object:nil];
+    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(authFailed:) name:@"registrationfailed" object:nil];
+    
+    [[NSNotificationCenter defaultCenter] removeObserver:self name:@"registrationdone" object:nil];
+    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(authCompleted:) name:@"registrationdone" object:nil];
+
+    [[NSNotificationCenter defaultCenter] removeObserver:self name:@"authfailed" object:nil];
+    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(authFailed:) name:@"authfailed" object:nil];
+    
+    [[NSNotificationCenter defaultCenter] removeObserver:self name:@"authdone" object:nil];
+    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(authCompleted:) name:@"authdone" object:nil];
+
+}
+
+
+-(void)viewWillDisappear:(BOOL)animated{
+    
+    [super viewWillDisappear:animated];
+    
+    [[NSNotificationCenter defaultCenter] removeObserver:self name:@"registerfailed" object:nil];
+    [[NSNotificationCenter defaultCenter] removeObserver:self name:@"registrationdone" object:nil];
+    [[NSNotificationCenter defaultCenter] removeObserver:self name:@"authfailed" object:nil];
+    [[NSNotificationCenter defaultCenter] removeObserver:self name:@"authdone" object:nil];
+
+}
+
+
+-(void)viewDidAppear:(BOOL)animated{
+    
+    [super viewDidAppear:animated];
+    
+}
+
+
+#pragma mark - app flow actions
+
+- (IBAction)btnAuthTapped:(id)sender{
+
+    NSLog(@"btnAuth");
+
+    if ([self checkProvidedValues] == true){
+        
+        [self showActivity];
+        
+        TASingleton *singleton = [[TASingleton alloc] init];
+        
+        if ([singleton isConnected] == true){
+            
+            NSString *userName = tfLogin.text;
+            NSString *password = tfPassword.text;
+            
+            BOOL modeLoginRegister = false;
+            modeLoginRegister = [swMode isOn];
+
+            if (modeLoginRegister == false){
+
+                [singleton authUser:userName password:password];
+
+            }else{
+
+                [singleton registerUser:userName password:password];
+
+            }
+            
+        }else{
+            
+            [self hideActivity];
+            
+        };
+        
+    }
+    
+}
+
+- (IBAction)swModeChanged:(id)sender{
+    
+    NSLog(@"swMode");
+
+    BOOL modeLoginRegister = false;
+    modeLoginRegister = [swMode isOn];
+    
+    if (modeLoginRegister == false){
+        //do login
+        
+        [btnAuth setTitle:@"LOG IN" forState:UIControlStateNormal];
+        
+    }else{
+        //do register
+
+        [btnAuth setTitle:@"REGISTER" forState:UIControlStateNormal];
+
+    }
+    
+}
+
+
+-(BOOL)checkLogin{
```

Comment:
The method name isn't descriptive enough to figure what it does. Supposedly it's the validation and thus should be `validateLogin`.